### PR TITLE
WIP: feat(integrations): Support inbound status sync core

### PR DESCRIPTION
**Pull request by macqueen**
- needs to be updated to use the new model
- needs tests (but i didn't want to add them until this is using the new model)

@lauryndbrown does this seem like it'll be useable for vsts?

### CODE REVIEW

File: src/sentry/integrations/issues.py [hunk 1]
```diff
@@ -143,3 +146,87 @@ def sync_status_outbound(self, external_issue, is_resolved, project_id, **kwargs
...
+                )
+                continue
+
+            if should_unresolve:
+                groups_to_unresolve.append(group)
+            elif should_resolve:
+                groups_to_resolve.append(group)
+
+        if groups_to_resolve:
+            updated_resolve = Group.objects.filter(
```
Comment (lauryndbrown) [inline src/sentry/integrations/issues.py]: The two sections with `groups_to_resolve` vs `groups_to_unresolve` should probably be the same function it's pretty similar code.  Something like:
``` 
def change_status(self, groups, status, activity_type):
     updated_groups = Group.objects.filter(
                id__in=[g.id for g in groups],
      ).exclude(
                status=status,
     ).update(
                status=status,
     )
    if updated_groups:
       for group in groups:
              Activity.objects.create(
                  project=group.project,
                  group=group,
                   type=activity_type,
               )
```
so when you call it all you'll have to do is:
```
if groups_to_resolve:
   self.change_status(groups_to_resolve, GroupStatus.RESOLVED, Activity.SET_RESOLVED
``` 
And similarly with unresolved.

File: src/sentry/integrations/issues.py [hunk 2]
```diff
@@ -143,3 +146,87 @@ def sync_status_outbound(self, external_issue, is_resolved, project_id, **kwargs
         Propagate a sentry issue's status to a linked issue's status.
         """
         raise NotImplementedError
+
+    def should_resolve(self, configuration, data):
```
Comment (lauryndbrown) [inline src/sentry/integrations/issues.py]: The `should_resolve` and `should_unresolve` methods have me a bit confused. Could you give me an example of how these would be used? Is there a reason they are separate methods? If so, what would be the differences between the two?

**Review by lauryndbrown — CHANGES_REQUESTED** [review]

**Comment by macqueen** [comment]
disclaimer, haven't run the latest update at all and need to add tests, so maybe don't look at this yet 🙂

**Comment by macqueen** [comment]
ok, @lauryndbrown added a couple tests and implemented your suggestions if you wanna take another look!

**Review by lauryndbrown — APPROVED** [review]
Looks good to me! Just have to get tests running.

**Pull request merged**